component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,220 @@
+                OAuth2TokenValidationRequestDTO.OAuth2AccessToken token = requestDTO.new OAuth2AccessToken();
+                token.setIdentifier(accessToken);
+                token.setTokenType(DPoPConstants.OAUTH_HEADER);
+                requestDTO.setAccessToken(token);
+
+                //TODO: If these values are not set, validation will fail giving an NPE. Need to see why that happens
+                OAuth2TokenValidationRequestDTO.TokenValidationContextParam contextParam = requestDTO.new
+                        TokenValidationContextParam();
+                contextParam.setKey("dummy");
+                contextParam.setValue("dummy");
+
+                OAuth2TokenValidationRequestDTO.TokenValidationContextParam[] contextParams = {contextParam};
+                requestDTO.setContext(contextParams);
+
+                OAuth2ClientApplicationDTO clientApplicationDTO = oAuth2TokenValidationService
+                        .findOAuthConsumerIfTokenIsValid
+                                (requestDTO);
+                OAuth2TokenValidationResponseDTO responseDTO = clientApplicationDTO.getAccessTokenValidationResponse();
+
+                getAuthenticationResult(authenticationResult, responseDTO, authorizationHeader,
+                        authenticationRequest);
+
+                String consumerKey = clientApplicationDTO.getConsumerKey();
+
+                authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
+
+                if (StringUtils.isNotEmpty(responseDTO.getAuthorizedUser())) {
+                    User user = new User();
+                    String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername
+                            (responseDTO.getAuthorizedUser());
+                    user.setUserName(UserCoreUtil.removeDomainFromName(tenantAwareUsername));
+                    user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(tenantAwareUsername));
+                    user.setTenantDomain(MultitenantUtils.getTenantDomain(responseDTO.getAuthorizedUser()));
+                    authenticationContext.setUser(user);
+                }
+
+                authenticationContext.addParameter(DPoPConstants.CONSUMER_KEY, consumerKey);
+                authenticationContext.addParameter(Constants.OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
+                authenticationContext.addParameter(Constants.OAUTH2_VALIDATE_SCOPE,
+                        AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                String serviceProvider = null;
+
+                try {
+                    serviceProvider =
+                            OAuth2Util.getServiceProvider(consumerKey).getApplicationName();
+                } catch (IdentityOAuth2Exception e) {
+                    String error = String.format("Error occurred while getting the Service Provider" +
+                            " by Consumer key: %s", consumerKey);
+                    log.error(error, e);
```

Comment:
Why are we not rethrowing this?

Reply:
unable to re throw due to the overridden method since we cannot change the method signature.